src/System.Management.Automation/engine/InternalCommands.cs

```diff
@@ -275,164 +495,204 @@ protected override void BeginProcessing()
                 args: emptyArray);
         }
 
-        /// <summary>
-        /// Execute the processing script blocks on the current pipeline object
-        /// which is passed as it's only parameter.
-        /// </summary>
-        /// <exception cref="ParseException">Could not parse script.</exception>
-        /// <exception cref="RuntimeException">See Pipeline.Invoke.</exception>
-        /// <exception cref="ParameterBindingException">See Pipeline.Invoke.</exception>
-        protected override void ProcessRecord()
+        private void ProcessPropertyAndMethodParameterSet()
         {
-            Dbg.Assert(ParameterSetName == "ScriptBlockSet" || ParameterSetName == "PropertyAndMethodSet", "ParameterSetName is neither 'ScriptBlockSet' nor 'PropertyAndMethodSet'");
+            _targetString = string.Format(CultureInfo.InvariantCulture, InternalCommandStrings.ForEachObjectTarget, GetStringRepresentation(InputObject));
 
-            switch (ParameterSetName)
+            if (LanguagePrimitives.IsNull(InputObject))
             {
-                case "ScriptBlockSet":
-                    for (int i = _start; i < _end; i++)
+                if (_arguments != null && _arguments.Length > 0)
+                {
+                    WriteError(GenerateNameParameterError("InputObject", ParserStrings.InvokeMethodOnNull,
```

Comment:
With parameters on multiple lines here, it might be worth chopping down to have each on its own line

Reply:
Outside scope of this work.  I don't intend to clean up all of the old foreach cmdlet code in this PR.  Please create a new issue.